apm-sniffer/apm-sdk-plugin/graphql-plugin/graphql-12.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/graphql/v12/GraphqlInterceptor.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.graphql.v12;
+
+import graphql.execution.ExecutionPath;
+import graphql.execution.ExecutionStrategyParameters;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GraphqlInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        ExecutionStrategyParameters parameters = (ExecutionStrategyParameters) allArguments[1];
+        if (parameters == null || parameters.getParent().getPath() != ExecutionPath.rootPath()) {
+            return;
+        }
+        objInst.setSkyWalkingDynamicField(System.currentTimeMillis());
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        ExecutionStrategyParameters parameters = (ExecutionStrategyParameters) allArguments[1];
+        if (parameters == null || parameters.getParent().getPath() != ExecutionPath.rootPath()) {
+            return ret;
+        }
+        String name = parameters.getField().getSingleField().getName();
+        long latency = System.currentTimeMillis() - (long) objInst.getSkyWalkingDynamicField();
+        String info = buildLogicEndpointTagInfo(name, latency, null);
+        AbstractSpan span = ContextManager.firstSpan();
+        if (span == null || !span.isEntry()) {
```

Comment:
How a GraphQL span could not be the first span? Or not an entry span? A little confused.

Reply:
i've used a localSpan to represent a Grqphql subquery